core/src/main/java/org/elasticsearch/common/Strings.java

```diff
@@ -840,9 +827,6 @@ public static String collectionToDelimitedString(Iterable<?> coll, String delim,
     }
 
     public static String collectionToDelimitedString(Iterable<?> coll, String delim, String prefix, String suffix, StringBuilder sb) {
-        if (Iterables.isEmpty(coll)) {
-            return "";
-        }
```

Comment:
No. Take a look at `SearchSlowLog#toString`. I think the old implementation here was wrong.